tests/python/gaia-ui-tests/gaiatest/tests/functional/dialer/test_duplicate_phone_number.py

```diff
@@ -0,0 +1,47 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import expected, By, Wait
+from gaiatest import GaiaTestCase
+from gaiatest.mocks.mock_contact import MockContact
+from gaiatest.apps.phone.app import Phone
+from gaiatest.apps.contacts.app import Contacts
+
+
+class TestDuplicatePhoneNumber(GaiaTestCase):
+
+    _iframe_locator = (By.CSS_SELECTOR, '.popupWindow.active iframe[data-url*="matching"]')
+    _merge_button = (By.ID, 'merge-action')
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+        
+        self.contact2 = MockContact(givenName='Test', tel=None)
+        self.data_layer.insert_contact(self.contact2)
+
+        self.contact = MockContact(givenName= 'Test', tel={
+            'type': 'Mobile',
+            'value': '123456789'})
+        self.data_layer.insert_contact(self.contact)
+
+    def test_duplicate_phone_number(self):
+        self.phone = Phone(self.marionette)
+        self.phone.launch()
+
+        number_to_add = self.contact['tel']['value']
+        self.phone.keypad.dial_phone_number(number_to_add)
+
+        add_number = self.phone.keypad.tap_add_contact()
+        add_to_existing_contact = add_number.tap_add_to_existing_contact()
+        add_to_existing_contact.wait_for_contacts(2)
+    
+        edit_contact = add_to_existing_contact.contacts[0].tap(return_class='EditContact')
+        edit_contact.tap_update(return_details=False)
```

Comment:
As we're likely going to add more steps in this test, I'd recommend to put the merge panel in a page class. The main reason is: A user sees that after he tapped on Update, he gets the merge panel. Everything related to switching or waiting for the panel to arrive are "implementation" details. Hence, it should belong in a page class.